geojson/src/commonMain/kotlin/org/maplibre/spatialk/geojson/serialization/PositionSerializer.kt

```diff
@@ -1,56 +1,45 @@
 package org.maplibre.spatialk.geojson.serialization
 
+import kotlinx.serialization.ExperimentalSerializationApi
 import org.maplibre.spatialk.geojson.Position
 import kotlinx.serialization.KSerializer
 import kotlinx.serialization.SerializationException
+import kotlinx.serialization.builtins.ListSerializer
+import kotlinx.serialization.builtins.serializer
 import kotlinx.serialization.descriptors.SerialDescriptor
-import kotlinx.serialization.descriptors.StructureKind
-import kotlinx.serialization.descriptors.buildSerialDescriptor
+import kotlinx.serialization.descriptors.listSerialDescriptor
 import kotlinx.serialization.encoding.Decoder
 import kotlinx.serialization.encoding.Encoder
-import kotlinx.serialization.json.JsonDecoder
-import kotlinx.serialization.json.JsonEncoder
-import kotlinx.serialization.json.add
-import kotlinx.serialization.json.buildJsonArray
-import kotlinx.serialization.json.double
-import kotlinx.serialization.json.jsonArray
-import kotlinx.serialization.json.jsonPrimitive
 
 /**
  * [KSerializer] implementation for implementations of the [Position] interface.
  * Serializes a Position down to an array of numbers as specified by GeoJSON.
- * This serializer only works for converting to and from JSON.
  * A position maps to `[longitude, latitude, altitude]`.
  *
  * A position's [altitude][Position.altitude] is only included in the array if it is not null.
  */
 public object PositionSerializer : KSerializer<Position> {
-    override val descriptor: SerialDescriptor
-        get() = buildSerialDescriptor("Position", StructureKind.LIST)
+    @OptIn(ExperimentalSerializationApi::class)
+    override val descriptor: SerialDescriptor =
+        listSerialDescriptor(Double.serializer().descriptor)
```

Comment:
is it correct to use a `List<Double>` descriptor instead of `Position` like before?

I'm not deeply familiar with the KSerializer API but this feels off